packages/react-components/src/components/VideoGallery.test.tsx

```diff
@@ -0,0 +1,194 @@
+    jest.spyOn(acs_ui_common, '_convertRemToPx').mockImplementation((rem: number) => {
+      return rem * 16;
+    });
+  });
+
+  test('should render local video tile in the grid alongside remote tiles', () => {
+    const localParticipant = createLocalParticipant({
+      userId: 'localUser',
+      videoStream: { isAvailable: false, renderElement: createVideoDivElement() }
+    });
+
+    const root = mountVideoGalleryWithLocalParticipant({ localParticipant });
+
+    root.setProps({ layout: 'default' });
+    const remoteParticipants = Array.from({ length: 10 }, () =>
+      createRemoteParticipant({
+        videoStream: { isAvailable: false, renderElement: createVideoDivElement() }
+      })
+    );
+    root.setProps({ remoteParticipants });
+    const gridLayout = root.find(GridLayout);
+    expect(
+      gridLayout
+        .find(VideoTile)
+        .findWhere((n) => n.prop('userId') === 'localUser')
+        .exists()
+    ).toBe(true);
+  });
+
+  test('should not have floating local video tile present', () => {
+    const localParticipant = createLocalParticipant({
+      videoStream: { isAvailable: false, renderElement: createVideoDivElement() }
+    });
+
+    const root = mountVideoGalleryWithLocalParticipant({ localParticipant });
+
+    root.setProps({ layout: 'default' });
+    const remoteParticipants = Array.from({ length: 10 }, () =>
+      createRemoteParticipant({
+        videoStream: { isAvailable: false, renderElement: createVideoDivElement() }
+      })
+    );
+    root.setProps({ remoteParticipants });
+    expect(root.find(_ModalClone).exists()).toBe(false);
+  });
+
+  test('should render all video tiles in the grid ', () => {
+    const localParticipant = createLocalParticipant({
+      videoStream: { isAvailable: false }
+    });
+
+    const root = mountVideoGalleryWithLocalParticipant({ localParticipant });
+
+    root.setProps({ layout: 'default' });
+    const remoteParticipants = Array.from({ length: 10 }, () =>
+      createRemoteParticipant({
+        videoStream: { isAvailable: false, renderElement: createVideoDivElement() }
+      })
+    );
+    root.setProps({ remoteParticipants });
+    expect(tileCount(root)).toBe(11);
+    expect(audioTileCount(root)).toBe(11);
+    expect(videoTileCount(root)).toBe(0);
+    expect(gridTileCount(root)).toBe(11);
+    expect(gridAudioTileCount(root)).toBe(11);
+    expect(gridVideoTileCount(root)).toBe(0);
+    expect(root.find(HorizontalGallery).exists()).toBe(false);
+  });
+
+  test('should render max allowed video tiles with streams in the grid ', () => {
+    const localParticipant = createLocalParticipant({
+      videoStream: { isAvailable: true, renderElement: createVideoDivElement() }
+    });
+    // Need to mock hook _useContainerWidth because the returned width is used by HorizontalGallery to decide how many
+    // tiles to show
+    jest.spyOn(responsive, '_useContainerWidth').mockImplementation(() => {
```

Comment:
Can we mock this at the top along with `_convertRemToPx`. It'll avoid having to mock for future tests, and keep all the mocking in one place.

It's also a very similar reason for mocking.